services/managers/market_manager.py

```diff
@@ -0,0 +1,145 @@
+    def check_username(username):
+        logger.debug("Checking alliance market username %s" % username)
+        cursor = connections['market'].cursor()
+        cursor.execute(marketManager.SQL_CHECK_USERNAME, [marketManager.__santatize_username(username)])
+        row = cursor.fetchone()
+        if row:
+            logger.debug("Found user %s on alliance market" % username)
+            return True
+        logger.debug("User %s not found on alliance market" % username)
+        return False
+
+    @staticmethod
+    def check_user_email(username, email):
+        logger.debug("Checking if alliance market email exists for user %s" % username)
+        cursor = connections['market'].cursor()
+        cursor.execute(marketManager.SQL_CHECK_EMAIL, [email])
+        row = cursor.fetchone()
+        if row:
+            logger.debug("Found user %s email address on alliance market" % username)
+            return True
+        logger.debug("User %s email address not found on alliance market" % username)
+        return False
+
+
+    @staticmethod
+    def add_user(username, email, characterid, charactername):
+        logger.debug("Adding new market user %s" % username)
+        plain_password = marketManager.__generate_random_pass()
+        hash = bcrypt.encrypt(plain_password, rounds=13)
+        hash_result = hash
+        rounds_striped = hash_result.strip('$2a$13$')
+        salt = rounds_striped[:22]
+        username_clean = marketManager.__santatize_username(username)
+        if marketManager.check_username(username)== False:
+            if marketManager.check_user_email(username, email) == False:
+                try:
+                    logger.debug("Adding user %s to alliance market" % username)
+                    cursor = connections['market'].cursor()
+                    cursor.execute(marketManager.SQL_ADD_USER, [username_clean, username_clean, email, email, salt,
+                                                                hash, characterid, charactername])
+                    return username_clean, plain_password
+                except:
+                    logger.debug("Unsuccessful attempt to add market user %s" % username)
+                    return None
+            else:
+                logger.debug("Alliance market email %s already exists Updating instead" % email)
+                username_clean, password = marketManager.update_user_info(username)
+                return username_clean, password
+        else:
+            logger.debug("Alliance market username %s already exists Updating instead" % username)
+            username_clean, password = marketManager.update_user_info(username)
+            return username_clean, password
+
+    @staticmethod
+    def disable_user(username):
+        logger.debug("Disabling alliance market user %s " % username)
+        cursor = connections['market'].cursor()
+        cursor.execute(marketManager.SQL_DISABLE_USER, [username])
+        return True

```

Comment:
this will literally always return true even if the user doesn't exist. not necessarily a problem I guess, but something to consider